mongo/integration/cursor_test.go

```diff
@@ -215,9 +268,29 @@ func tryNextGetmoreError(mt *mtest.T, cursor tryNextCursor) {
 
 	err := cursor.Err()
 	assert.NotNil(mt, err, "expected change stream error, got nil")
+
+	// make sure that a mongo.CommandError is returned instead of a driver.Error
+	mongoErr, ok := err.(mongo.CommandError)
+	assert.True(mt, ok, "expected mongo.CommandError, got: %T", err)
+	assert.Equal(mt, testErr.Code, mongoErr.Code, "expected code %v, got: %v", testErr.Code, mongoErr.Code)
+	assert.Equal(mt, testErr.Message, mongoErr.Message, "expected message %v, got: %v", testErr.Message, mongoErr.Message)
+	assert.Equal(mt, testErr.Name, mongoErr.Name, "expected name %v, got: %v", testErr.Name, mongoErr.Name)
+	assert.True(mt, sliceStringEqual(testErr.Labels, mongoErr.Labels), "expected labels %v, got: %v", testErr.Labels, mongoErr.Labels)
 }
 
 func assertCursorBatchLength(mt *mtest.T, cursor *mongo.Cursor, expected int) {
 	batchLen := cursor.RemainingBatchLength()
 	assert.Equal(mt, expected, batchLen, "expected remaining batch length %d, got %d", expected, batchLen)
 }
+
+func sliceStringEqual(a []string, b []string) bool {
```

Comment:
Is this necessary? Our assertion library internally calls `cmp.Equal` to check equality and the [documentation](https://godoc.org/github.com/google/go-cmp/cmp#Equal) for that function says:

> Slices are equal if they are both nil or both non-nil, where recursively calling Equal on all non-ignored slice or array elements report equal. Empty non-nil slices and nil slices are not equal; to equate empty slices, consider using cmpopts.EquateEmpty.

To me, this implies that we should be able to do `assert.Equal(mt, slice1, slice2)`.